src/hooks/misc/serialize_join_request_message.cpp

```diff
@@ -12,7 +13,7 @@ namespace big
 		if (info->unk_0xC0 == 0)
 			info->unk_0xC0 = 1; // TODO: do we need this anymore?
 
-		if (g.protections.desync_kick)
+		if (g.protections.desync_kick && (!*g_pointers->m_gta.m_is_session_started || !g_player_service->get_self()->is_host()))

```

Comment:
serialize_join_request_message & serialize_join_request_message2 hooks are both called once before joining session like-

```
[15:29:14.9634776][DEBUG/player_service.cpp](65:4) Created self player
[15:29:56.9487358][INFO/network_player_mgr.cpp](41:3) CNetworkPlayerMgr init
[15:29:57.0869530][INFO/serialize_join_request_message.cpp](25:3) serialize_join_request_message 2
[15:29:57.0899135][DEBUG/player_service.cpp](65:4) Created self player
[15:29:58.2004453][INFO/serialize_join_request_message.cpp](13:3) serialize_join_request_message
[15:30:00.0933630][INFO/assign_physical_index.cpp](43:4) You joined the session.
[15:30:00.1181372][INFO/assign_physical_index.cpp](93:7) Player joined: 'Molot-OvO'(host), slot #0, RID: 219592377, tk: 9262475521841728060
[15:30:00.7425081][INFO/assign_physical_index.cpp](93:7) Player joined: 'SwanLiHong', slot #22, RID: 188695962, tk: 17452440618492821689
[15:30:00.8407558][INFO/assign_physical_index.cpp](93:7) Player joined: 'TuTuGTAV6', slot #23, RID: 234423164, tk: 16175699075157837359
[15:30:00.9668756][INFO/assign_physical_index.cpp](93:7) Player joined: 'LSPZIO2333', slot #8, RID: 110405142, tk: 15216796575502584149
```

if host broadhost our details then our `nat_type=0` will be always broadcasted to others. How these checks help bcz the hooks are only called once and most probably these checks either will not set nat to 0 or will set nat to 0. So there will 50% probability that yim is detected with desync protection

Reply:
dont have idea for sure but just thinking out of brain. May close this if not concerning,

Reply:
i mean the checks in `assign_physical_index` will be always add infraction to yim users that they have desync protection

Reply:
serialize_join_request_message and serialize_join_request_message2 hooks are called before we join session and then other players who see us there assign_physical_index hooks will trigger that we have desync protection thats what i meant actually

Reply:
> They are also called when you're the host and are sending player data to other clients.

so we will always have our nat type spoofed to 0 when not host.

> some menus even play with the NAT types of other clients for absolutely no reason

may be to protect everyone from desync kicks

> Make our protection harder to detect

this will only work well when we host the session from starting?